lib/rubygems/source_local.rb

```diff
@@ -2,5 +2,4 @@
 require 'rubygems/source'
 require 'rubygems/source_local'
 
-# TODO warn upon require, this file is deprecated.
-
+Kernel.warn "Requiring rubygems/source_local is deprecated; please use rubygems/source/local instead."
```

Comment:
@segiddins I attempted to make a new version of this PR that uses `Gem.location_of_caller`, but...

in source_local.rb:

```
  Kernel.warn "#{Gem.location_of_caller.join(':')}: Warning: Requiring rubygems/source_local is deprecated; please use rubygems/source/local instead."
```

x.rb:

```ruby
require 'rubygems/source_local'
```

`ruby -Ilib x.rb` output:

```
/home/ellen/dev/ruby/rubygems/rubygems/lib/rubygems/core_ext/kernel_require.rb:54: Warning: Requiring rubygems/source_local is deprecated; please use rubygems/source/local instead.
```

am I missing something?